cloudify_aws/common/decorators.py

```diff
@@ -106,12 +106,16 @@ def wrapper_inner(**kwargs):
             ctx = kwargs['ctx']
             # Add new operation arguments
             kwargs['resource_type'] = resource_type
-            kwargs['iface'] = class_decl(
-                ctx.source.node, logger=ctx.logger,
-                resource_id=utils.get_resource_id(
-                    node=ctx.source.node,
-                    instance=ctx.source.instance,
-                    raise_on_missing=True)) if class_decl else None
+            iface = kwargs.get('iface')
```

Comment:
Because in this code we start creating iface of one type from another, for example:
```
        iface = EC2TransitGatewayAttachment(
            ctx.source.node,
            transit_gateway_attachment_id,
            iface.client,
            ctx.logger)
```
This makes the tests simpler. And this is also just better code than previous implementation.

Reply:
Cool